src/components/Modals/Receive/Receive.tsx

```diff
@@ -11,97 +11,157 @@ import {
   ModalFooter,
   ModalHeader,
   ModalOverlay,
-  useColorModeValue
+  useColorModeValue,
+  useToast
 } from '@chakra-ui/react'
+import { ChainIdentifier } from '@shapeshiftoss/chain-adapters'
+import { AssetMarketData } from '@shapeshiftoss/market-service'
 import { Card } from 'components/Card'
 import { QRCode } from 'components/QRCode/QRCode'
 import { RawText, Text } from 'components/Text'
+import { useChainAdapters } from 'context/ChainAdaptersProvider/ChainAdaptersProvider'
 import { useModal } from 'context/ModalProvider/ModalProvider'
+import { useWallet } from 'context/WalletProvider/WalletProvider'
+import { useEffect, useState } from 'react'
 import { useTranslate } from 'react-polyglot'
 
-const Receive = () => {
+type ReceivePropsType = {
+  asset: AssetMarketData
+}
+
+export const ReceiveModalProps: ReceivePropsType = {
```

Comment:
this is my only concern about the way i refactored the modal provider - it might not be feasible to pass in valid props to all modals we want to use, but i think i have another way to do this while strongly typing the props based on the modal.